src/modules/subject/subjects.routes.js

```diff
@@ -0,0 +1,23 @@
+const router = require('express').Router()
+const subjectController = require('./subject.controller')
+const asyncWrapper = require('~/middlewares/asyncWrapper')
+const idValidation = require('~/middlewares/idValidation')
+const { restrictTo, authMiddleware } = require('~/middlewares/auth')
+
+const {
+  roles: { ADMIN }
+} = require('~/consts/auth')
+
+router.use(authMiddleware)
+router.param('id', idValidation)
+
+router.get('/', asyncWrapper(subjectController.subjectsFind))
+router.get('/:id', asyncWrapper(subjectController.subjectFindById))
+router.get('/categories/:id', asyncWrapper(subjectController.subjectsFindByCategoryId))
```

Comment:
We discussed that this route should be like `/categories/:id/subjects` , not `subjects/categories/:id`.